src/runtime.js

```diff
@@ -11,11 +11,43 @@ Component.prototype.render = function render() {
   throw new Error('Blop Component need to implement the render method');
 };
 
-function createComponent(Comp, attributes, children) {
+// this global is an issue you mount
+// several time
+let globalRefresh = null;
+let currentNode = null;
+// todo: garbage collect the state cache?
+// this is the component state cache
+const cache = {};
+
+function useState(initialValue) {
+  const { state, currentState } = currentNode;
+  currentNode.state[currentState] = state[currentState] || initialValue;
+  const setStateHookIndex = currentState;
+  const setState = (newState) => {
+    state[setStateHookIndex] = newState;
+    globalRefresh();
+  };
+  currentNode.currentState = currentState + 1;
+  return { value: state[currentState], setState };
+}
+
+function createComponent(Comp, attributes, children, name) {
+  const path = currentNode ? `${currentNode.path}.${currentNode.children.length}.${name}` : name;
```

Comment:
This path is the way I found to locate the previously saved state from a previous render pass... I do not know how React handle this...